src/transformers/models/deprecated/open_llama/modeling_open_llama.py

```diff
@@ -578,6 +578,13 @@ def forward(
         seq_length_with_past = seq_length
         past_key_values_length = 0
 
+        if self.gradient_checkpointing and self.training:
```

Comment:
I don't see why moving the logic here should make a difference?

Reply:
It is very unusual but apparently when using GC + training the backward pass calls again the forward pass but with a non-`None` `past_key_value`. 
I think `use_cache` is set to `True` by default on all configs; therefore avoiding the block
```python
        if use_cache:
            use_legacy_cache = not isinstance(past_key_values, Cache)
            if use_legacy_cache:
                past_key_values = DynamicCache.from_legacy_cache(past_key_values)
            past_key_values_length = past_key_values.get_usable_length(seq_length)
```
to be called seems to be the right fix. I will add tests directly on PEFT as it is related to PEFT